### ISSUE TITLE
How to train with large datasets?

### ISSUE THREAD

**Issue by manglav** [KindXiaoming/pykan#122]
I am trying to use a massive CSV (1M+ rows), with the X input having 400 dimensions. It's technically working but I had two main questions: 

1. It seems like the GPU is bottlenecked a bit by the CPU - I'm only seeing 35% GPU usage vs 100% single core CPU. If I run in CPU mode, I see around 22% CPU (across 64 cores). Any suggestions on improving this?

2. The data is so large that we are reading the csv in chunks of 1000 and calling train. Is this the right approach? What is the best way to train with this big data? What about the  <bias_trainable, sp_trainable, and sb_trainable> parameters?

```
model = KAN(width=[400,64, 32, 16,1], grid=10, k=4, seed=0, device=device, bias_trainable=False, sp_trainable=False, sb_trainable=False)
for chunk in chunks:
...< data processing >....
data_dict = create_data_from_chunk(chunk)
model.train(data_dict, opt="LBFGS", steps=1, device=device, update_grid=False)
```

3. I can save a checkpoint, but I don't know how to use the trained model either. Is there a "predict" function to use?

**Comment by JinglaiZheng** [comment]
I had the same problem, I couldn't find the API for prediction after training the model.

**Comment by Zhangyuyi-0825** [comment]
> I had the same problem, I couldn't find the API for prediction after training the model.

My research is in the area of time series forecasting (tabular data), where the training result of the model is a mathematical expression, and therefore its prediction results are obtained by bringing the data from the test set into the expression.
However, the forecasting performance is very, very, very unsatisfactory and I am still trying to adjust the network structure and parameters.

**Comment by ChrisD-7** [comment]
building on the GPU training I tried running it on my CPU and ran out of compute when running it with GPU on colab for the device.ipynb code got this error

![image](https://github.com/KindXiaoming/pykan/assets/157317426/42444614-de5b-499a-a957-5fc141c3c1c7)

**Comment by seyidcemkarakas** [comment]
> I had the same problem, I couldn't find the API for prediction after training the model.

Did you handle this situation?
